app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -150,8 +153,35 @@ public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float d
 
     @Override
     public void onLongPress(MotionEvent e) {
-        // Also switch on long-press for accessibility
-        setMainImage(true, true);
+        openCurrentMainImageInGallery();
+    }
+
+    private void openCurrentMainImageInGallery() {
+        ImageType wantedImageType = imageTypes.get(mainImageIndex);
+
+        File file = null;
+        if (wantedImageType == ImageType.IMAGE_FRONT) {
+            file = Utils.retrieveCardImageAsFile(this, loyaltyCardId, ImageLocationType.front);
+        }
+        else if (wantedImageType == ImageType.IMAGE_BACK) {
+            file = Utils.retrieveCardImageAsFile(this, loyaltyCardId, ImageLocationType.back);
+        }
+
+        // Entry is either the barcode (not an image) or image retrieval failed
+        if (file == null) {
+            return;
+        }
+
+        try {
+            Intent intent = new Intent(Intent.ACTION_VIEW)
+                    .setDataAndType(FileProvider.getUriForFile(this, BuildConfig.APPLICATION_ID, file), "image/*")
+                    .addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+            startActivity(intent);
+        }
+        catch (ActivityNotFoundException e) {
+            // Do nothing if an image viewer is not installed on device except logging
+            e.printStackTrace();
```

Comment:
Oh, I completely forgot the ActivityNotFound could happen. Thank you for adding this!

Could it be possible to add a toast here using the `failedLaunchingPhotoPicker` string, like is used in ScanActivity and LoyaltyCardEditActivity? (Hmm, that string kinda has a bad name but it contains the right message, so whatever :sweat_smile:)